### PR TITLE
feat: add `nonce` support

### PR DESCRIPTION
**Pull request by trijpstra-fourlights**
Implements #1011

### CODE REVIEW

File: docs/content/2.usage/1.nuxt-img.md [hunk 1]
```diff
@@ -269,6 +269,20 @@ Set `loading="lazy"` to defer loading of an image until it appears in the viewpo
...
+### `nonce`
+
+This is a [native](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce) global attribute that defines a cryptographic nonce (number used once) that can be used by Content Security Policy to determine whether or not a given fetch will be allowed to proceed for a given element.
+Providing a nonce allows you to avoid using the CSP `unsafe-inline` directive, which would allowlist _all_ inline script or styles.
+
+```html
+<NuxtImg src="/nuxt-icon.png" :nonce="nonce" />
+
+<script lang="ts" setup>
+const nonce = useNonce() // using `nuxt-security` module
```
Comment (danielroe) [inline docs/content/2.usage/1.nuxt-img.md:281]:
```suggestion
// useNonce is not provided by nuxt/image but might be
// provided by another module, for example nuxt-security
const nonce = useNonce()
```

File: src/runtime/components/nuxt-img.ts [hunk 2]
```diff
@@ -78,6 +78,7 @@ export default defineComponent({
       useHead({
         link: [{
           rel: 'preload',
+          nonce: props.nonce,
```
Comment (danielroe) [inline src/runtime/components/nuxt-img.ts:81]:
```suggestion
          ...(props.nonce ? { nonce: props.nonce } : {}),
```

Comment (trijpstra-fourlights) [reply src/runtime/components/nuxt-img.ts]: I see you removed this quite quickly after my recent push in https://github.com/nuxt/image/pull/1012/commits/d1d5258b9711f8669b70ef94e5c1d32987e4d90b 

Locally `pnpm test:types` (on https://github.com/nuxt/image/pull/1012/commits/07b141e0d82b742ed81f908911af898a819bee48) runs without error (and without the need for the `ts-expect-error`).

Comment (danielroe) [reply src/runtime/components/nuxt-img.ts]: sorry, I was editing locally at the same time 🙈

the reason I prefer this is that the spread _does_ work without type errors, but it also removes type safety here as you could pass anything in, e.g. `{ bob: true }`.

Comment (trijpstra-fourlights) [reply src/runtime/components/nuxt-img.ts]: I see, makes sense! Thanks for all your hard work on nuxt btw :)

**Review by Baroshem — APPROVED** [review]

File: src/runtime/components/nuxt-picture.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import { h, defineComponent, ref, computed, onMounted } from 'vue'
+import type { Head } from 'zhead'
```
Comment (harlan-zw) [inline src/runtime/components/nuxt-picture.ts:2]: zhead is an implicit dependency, better to use the explicit dependency 

```ts
import type { Head } from '@unhead/vue'
```

File: src/runtime/components/nuxt-img.ts [hunk 4]
```diff
@@ -79,6 +79,8 @@ export default defineComponent({
         link: [{
           rel: 'preload',
           as: 'image',
+          // @ts-expect-error might need to resolve upstream in unhead
```
Comment (harlan-zw) [inline src/runtime/components/nuxt-img.ts:82]: This is fixed in zhead [v2.1.2](https://github.com/harlan-zw/zhead/releases/tag/v2.1.2)

Comment (trijpstra-fourlights) [reply src/runtime/components/nuxt-img.ts:82]: In the context of this PR, what's the preferred resolution here? 

Wait for the new release of `@unhead/vue` to use `zhead@^2.1.2`? Or do you want me to update the comment here to something like `fixed in zhead v2.1.2, needs upstream adoption by @unhead/vue`?

Comment (harlan-zw) [reply src/runtime/components/nuxt-img.ts:82]: Good question, if you delete the lock file & node_modules it should update to this version and you can remove the comment altogether

Comment (danielroe) [reply src/runtime/components/nuxt-img.ts:82]: don't worry about refreshing lockfile in this pr; I can remove the comment when we next update deps....

**Review by danielroe — APPROVED** [review]

**Review by harlan-zw — APPROVED** [review]

**Comment by Baroshem** [comment]
Awesome! 🚀

**Pull request merged**